bbq/vm/test/vm_test.go

```diff
@@ -7845,3 +7845,111 @@ func TestSaturatingArithmetic(t *testing.T) {
 		result,
 	)
 }
+
+func TestGlobalVariables(t *testing.T) {
```

Comment:
Could you please add some tests that have side effects, like calling a function (e.g. logging), and assert the evaluation order?

Reply:
Added in https://github.com/onflow/cadence/pull/3920/commits/37ea957ecec57cc9a7c9cd39c69b64dbbef7073d